lib/publiccloud/azure.pm

```diff
@@ -82,7 +83,8 @@ sub vault_create_credentials {
 
 sub resource_exist {
     my ($self) = @_;
-    my $output = script_output(q(az group list --query "[?name=='openqa-upload']"));
+    my $group  = $self->resource_group;
+    my $output = script_output_retry("az group show --name '$group' --output json", retry => 3, timeout => 30, delay => 10);
```

Comment:
```suggestion
    return script_output_retry("az group show --name '$group' --output json", retry => 3, timeout => 30, delay => 10);
```
this `az group show` is really much better here :+1:.
But it return with `3` if the resource group doesn't exists.
```
clemix@linux-7k5u:~/sandbox/perl> az group show --name 'donexists' --output json
Error loading command module 'taskhelp'
Resource group 'donexists' could not be found.
clemix@linux-7k5u:~/sandbox/perl> echo $?
3
```

Reply:
In that case the test will fail, because `script_output_retry` only returns the output, then the command returns with `0`. We are using `script_output` internally, which, AFAICT throws an error when `proceed_on_failure => 0`.
Isn't that what we want to have?

Reply:
hm and with this we might break the logic of: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure.pm#L92

Sorry for late notice!

Reply:
The VR for `upload_image` succeeds though: http://phoenix-openqa.qam.suse.de/tests/3874

Reply:
Yeah ok and we would just `die()` which is actually ok as well